### PR TITLE
AI Chat Entry point B - Address bar

### PR DESCRIPTION
**Pull request by Bunn**
Task/Issue URL: https://app.asana.com/0/1204167627774280/1208794359260769/f

**Description**:
Add AI Chat to the Omnibar accessory button

**Steps to test this PR**:
1. Make sure you're set as internal user
2. Go to a SERP, check that instead of the share button you'll see the chat button
3. Check if the chat button works
4. Open any site from the SERP, check if the button changes to the share button
5. Check if the share button works
6. Go to settings, disable the AI Chat in search results
7. Do the same steps above, make sure we never display the chat button

----
1. Do a clean install, do not make yourself an internal user.
2. Validate that you do not see the chat icon there

### CODE REVIEW

File: DuckDuckGo/MainViewController.swift [hunk 1]
```diff
@@ -1544,6 +1556,15 @@ class MainViewController: UIViewController {
             .store(in: &settingsDeepLinkcancellables)
     }
 
+    private func subscribeToAIChatSettingsEvents() {
+        NotificationCenter.default.publisher(for: .aiChatSettingsChanged)
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] notification in
```
Comment (brindy) [inline DuckDuckGo/MainViewController.swift]: SwiftLint is complaining about the wrong line ☝️

File: DuckDuckGo/LargeOmniBarState.swift [hunk 2]
```diff
@@ -27,7 +27,7 @@ struct LargeOmniBarState {
         let showBackButton: Bool = true
         let showForwardButton: Bool = true
         let showBookmarksButton: Bool = true
-        let showShareButton: Bool = false
+        let shoeAccessoryButton: Bool = false
```
Comment (brindy) [inline DuckDuckGo/LargeOmniBarState.swift]: typo `shoe` -> `show`

File: DuckDuckGo/OmniBarDelegate.swift [hunk 3]
```diff
@@ -92,7 +92,7 @@ extension OmniBarDelegate {
         
     }
 
-    func onShareLongPressed() {
+    func onShareLongPressed(accessoryType: OmniBar.AccessoryType) {
```
Comment (brindy) [inline DuckDuckGo/OmniBarDelegate.swift]: Rename this to onAccessoryLongPressed?

**Review by brindy — CHANGES_REQUESTED** [review]
looks good, but a few minor things...

* rename onShareLongPress -> onAccessoryLongPress for consistency
* SwiftLint (was flagging the wrong line 🙄 )
* typo of the showAccessoryButton var in the omni bar state

File: DuckDuckGo/LargeOmniBarState.swift (hunk 2 above)
Comment (Bunn) [reply DuckDuckGo/LargeOmniBarState.swift]: It's a shoe 🥿  accessory 😬 
Nice catch, thanks!

File: DuckDuckGo/OmniBarDelegate.swift (hunk 3 above)
Comment (Bunn) [reply DuckDuckGo/OmniBarDelegate.swift]: Yep, oversight, thanks!

**Comment by Bunn** [comment]
@brindy I've addressed all the suggestions, thanks for it! 
As we discussed on MM, here are some additional changes:
* Added a conditional UI for settings using the feature flags and also fixed some settings issues -> https://github.com/duckduckgo/iOS/pull/3694/commits/96e1e5ab9c96706af20fb4070a02fee08f7d3c1a 
* New tests related to the issues fixed above -> https://github.com/duckduckgo/iOS/pull/3694/commits/07c80c8457b1c5a044eadd6f2ea6f5ee552bfd28
* Small rename to make the property names more consistent -> https://github.com/duckduckgo/iOS/pull/3694/commits/c492804718b0e19bffb12ad134c27baf116bb48a
* Fix that issue related to not dismissing when opening a not-AI Chat link -> https://github.com/duckduckgo/iOS/pull/3694/commits/8c25e8df59a4a31abc4287848cc3ea766ec6b4a7

If you want to test different privacy configs:
* http://jsonblob.com/api/1315696304357302272 browsing menu enabled/ address bar disabled

* http://jsonblob.com/api/1315696595987259392 browsing menu disabled / address bar enabled

* http://jsonblob.com/api/1315697149798965248 all enabled

**Review by brindy — APPROVED** [review]
LGTM!

**Pull request merged**